### ISSUE TITLE
[Server Crash] TypeError: Cannot read properties of null (reading 'once') at internals.timeout (...)

### ISSUE THREAD

**Issue by chirigg** [senecajs/seneca-transport#181]
# Hi!

There is an issue accessing socket events when there are concurrent connections and seneca is called via https with no parameters, causing the server to crash. At the moment I could only replicate the problem under these conditions.

## Steps to replicate
1. Fork `seneca-transport`
2. Modify `./integration/server.js` with
```
"use strict";

var Transport = require("../../transport");
var Seneca = require("seneca");

var color = function () {
  this.add("color:red", function (args, callback) {
    callback(null, { hex: "#FF0000" });
  });
};

const fs = require("node:fs");

const httpsOptions = {
  port: 8000,
  protocol: "https",
  serverOptions: {
    key: fs.readFileSync("/tmp/SenecaKey.key", "utf8"),
    cert: fs.readFileSync("/tmp/SenecaCertificate.crt", "utf8"),
  },
};

Seneca({ default_plugins: { transport: false } })
  .use(Transport)
  .use(color)
  .listen(httpsOptions);
```
3. Generate the certificate `openssl genrsa -out /tmp/SenecaKey.key 2048
openssl req -new -x509 -key /tmp/SenecaKey.key -out /tmp/SenecaCertificate.crt -days 365`
4. `brew install wrk`
5. Run the server `node ./test/integration/server.js`
6. Run the test `wrk -t1 -c1 -d10s "https://localhost:8000/act"`
7. You will get the following error `TypeError: Cannot read properties of null (reading 'once') at internals.timeout (...)`

# Candidate fix
In the file `lib/http.js`, L360 (https://github.com/senecajs/seneca-transport/blob/master/lib/http.js#L360) replace 
```
res.socket.once('data', clearTimeoutId)
```
with
```
if(res.socket) {
  res.socket.once("data", clearTimeoutId);
} else {
  clearTimeoutId();
}
```
In this way we ensure that the timeout is handled appropriately in the event that the socket does not exist.

# Results
With the fix
```
wrk -t1 -c1 -d10s "https://localhost:8000/act"

Running 10s test @ https://localhost:8000/act
  1 threads and 1 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   287.89us  299.18us   6.41ms   93.75%
    Req/Sec     3.99k   534.63     4.51k    82.00%
  39639 requests in 10.00s, 36.10MB read
  Non-2xx or 3xx responses: 39639
Requests/sec:   3963.28
Transfer/sec:      3.61MB
```

Without the fix (Server Crash)
```
wrk -t1 -c1 -d10s "https://localhost:8000/act"

Running 10s test @ https://localhost:8000/act
  1 threads and 1 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     2.09ms    2.27ms   8.11ms   88.89%
    Req/Sec    90.00      0.00    90.00    100.00%
  9 requests in 10.00s, 8.39KB read
  Socket errors: connect 130824, read 1, write 0, timeout 0
  Non-2xx or 3xx responses: 9
Requests/sec:      0.90
Transfer/sec:     859.07B
```
# PR
https://github.com/senecajs/seneca-transport/pull/182

**Comment by chirigg** [comment]
@rjrodger what do you think 🧐?

**Comment by rjrodger** [comment]
thank you! merged and published! 8.3.0
